packages/applicant/src/pages/mandatory-questions/[mandatoryQuestionId]/getServerSideProps.tsx

```diff
@@ -102,6 +142,20 @@ export default async function getServerSideProps({
     defaultFields = response.body as Optional<GrantMandatoryQuestionDto>;
   }
 
+  let backButtonUrl = '';
+  if (fromSubmissionPage) {
+    backButtonUrl = routes.submissions.section(submissionId, sectionId);
+  } else if (fromSummaryPage) {
+    backButtonUrl = routes.mandatoryQuestions.summaryPage(mandatoryQuestionId);
+  } else {
+    const mapper = buildBackButtonMapper(
+      mandatoryQuestion.orgType,
+      mandatoryQuestionId,
+      mandatoryQuestion.schemeId
+    );
+    backButtonUrl = mapper[resolvedUrl];
+  }
```

Comment:
This `getServerSideProps` is well long already, could stick this in a function as well

Reply:
can do, unfortunately it barely saves any lines because of how many parameters the function needs, but at least reduces getServerSideProps complexity